biz.aQute.bndlib/src/aQute/bnd/osgi/Processor.java

```diff
@@ -2946,4 +2948,51 @@ public synchronized String getProfile() {
 		}
 		return profile;
 	}
+
+	/**
+	 * A checksum based on the values of the properties
+	 *
+	 * @return A checksum based on the values of the properties
+	 */
+	public String getChecksum() {
+		try {
+			Properties properties2 = getProperties();
+			String old = properties2.getProperty(Constants.TSTAMP);
+			properties2.setProperty(Constants.TSTAMP, "0");
```

Comment:
This seems risky from a thread safety point of view. Isn't is better an cleaner to simply set 

`flattenedProperties.setProperty(Constants.TSTAMP, "0");`

after creating the flattened properties object?

Reply:
Problem is macro expansion. Lots of macros tstamp

Reply:
Perhaps then narrow the window and restore TSTAMP right after you flatten the properties?

Reply:
I've replaced this with a new processor that inherits this processor. Checking if this works

Reply:
This works well, and is quite clean